authority/config/config.go

```diff
@@ -106,6 +109,12 @@ type AuthConfig struct {
 	DisableGetSSHHosts   bool                  `json:"disableGetSSHHosts,omitempty"`
 }
 
+// CRLConfig represents config options for CRL generation
+type CRLConfig struct {
+	Generate      bool                  `json:"generate,omitempty"`
+	CacheDuration *provisioner.Duration `json:"cacheDuration,omitempty"`
+}
```

Comment:
The configuration here is weird, First, I fixed a panic with `"crl": {}`, on revocation there was a panic on `GenerateCertificateRevocationList` because  `CacheDuration` was nil. Then if generate is false, the CRL won't be renewed unless a certificate is revoked. So you will be able to get a CRL, but it will be expired.

Reply:
This weirdness is to support multiple nodes:
- One node should have `generate: true` so that a CRL gets generated regularly
- Additional nodes should have `cacheDuration: XXm` and `generate: false` so that if a revoke request is routed to the extra nodes, it can successfully generate a CRL and place it in the DB with the correct duration set

I'll add some logic to test for 0-length durations though, unless you would like the config refactored? It is a bit weird, so im open to suggestions

Reply:
The second node doesn't need to create a CRL to store the proper `ExpiresAt` in the DB, what feels weird is that it will generate a CRL that will expire and won't be regenerated unless another revoke is routed to that node.

What makes sense to me is having one node that generates every XX time (24h by default) a new CRL, and perhaps optionally regenerates it if a new revoke occurs. Although this is not common as clients can have in their cache the old CRL because it hasn't expired.

Reply:
I'm a little confused, perhaps i've misunderstood something somewhere.

The multiple nodes scenario i'm looking at is the following:
* `Node A` connected to `SharedDB` sitting behind `LoadBalancer`
* `Node B` connected to `SharedDB` sitting behind `LoadBalancer`

Lets assume `Node A` has `crl: {"generate": true, "cacheDuration": "5m"}`

`Node A` starts up and generates a CRL regularly, storing it in `SharedDB`. `ClientA` requests the CRL regularly from the endpoint, the request being routed randomly between `Node A` and `Node B`.

`Client B` sends a revoke request, and it is routed the `Node B`. `Node B` will add the revoked cert to the list of revoked certificates in `SharedDB` and initiate regeneration of the CRL, and in doing so needs to place a value into the `NextUpdate` field in the CRL. Where should `Node B` get the duration for calculating `NextUpdate` from? At present `Node B` uses `cacheDuration` to calculate it. Perhaps it could be calculated from the old CRL instead?

In this scenario im assuming we only want one node to be regularly generating CRLs, so one node has `generate: true` while all others have it set to `false`

I think there is also a concurrency issue where `Node A` could be regenerating the CRL while `Node B` is handling a revoke request, but im presuming that could be handled by the database?

Of course its entirely possible I have misunderstood your point here, so let me know if thats the case :)